x-pack/qa/repository-old-versions/src/test/java/org/elasticsearch/oldrepos/OldRepositoryAccessIT.java

```diff
@@ -193,7 +211,19 @@ public void testOldRepoAccess() throws IOException {
                     restoreMountAndVerify(numDocs, expectedIds, client, numberOfShards);
                 }
             } finally {
-                oldEs.performRequest(new Request("DELETE", "/test"));
+                IOUtils.closeWhileHandlingException(
+                    () -> oldEs.performRequest(new Request("DELETE", "/test")),
+                    () -> oldEs.performRequest(new Request("DELETE", "/_snapshot/testrepo/snap1")),
+                    () -> oldEs.performRequest(new Request("DELETE", "/_snapshot/testrepo"))
+                );
+                if (Build.CURRENT.isSnapshot()) {
```

Comment:
The functionality to read older indices is currently only available in snapshot builds.